src/sail_json_backend/json.ml

```diff
@@ -239,29 +239,31 @@ let filter_non_operands components =
   in
   aux [] components
 
-let extract_operands k regex components =
+let extract_operands k components =
   let rec aux acc = function
     | [] -> List.rev acc
     | hd :: tl ->
-        if Str.string_match regex hd 0 then (
+        if Str.string_match (Str.regexp ".+(\\(.*\\))") hd 0 then (
           let operand = Str.matched_group 1 hd in
+          let elements = Str.split (Str.regexp ",") operand in
           try
-            let comma_index = String.index operand ',' in
-            debug_print ("Operand before trimming: " ^ operand);
-            let trimmed = String.sub operand 0 comma_index in
-            debug_print ("Final trimmed operand: " ^ trimmed);
             let inputl = Hashtbl.find inputs k in
             if List.mem trimmed inputl then aux (trimmed :: acc) tl else aux acc tl
```

Comment:
You removed `trimmed` above, so I'm not sure how you can use it here? :-)

Reply:
Fixed : )